src/test/java/collectionsEducation/CountryWorker.java

```diff
@@ -4,88 +4,189 @@
+    private static List<String> getSortListOfNames(List<String> names, boolean isAscendingOrder) {
         List<String> filteredCountries = new ArrayList<>();
-//    Замість мого коментаря повинна бути реалізація
+
+                 if (isAscendingOrder) {
+                      Collections.sort(names);//приходить тру
+                 } else {
+                      Collections.sort(names,Collections.reverseOrder());//приходить фолс
+                 }
+                 //    Замість мого коментаря повинна бути реалізація
 //    Задача 3: відсортіруй імена в алфавітному порядку,
 //    в залежності від значення isAscendingOrder сортуй або від А-Я, або Я-А
 //    Сортування теж Case insensitive
-        return filteredCountries;
+        return names;
     }
 
-    private List<CountryDTO> getListOfCountriesNameEquals(List<CountryDTO> allCountries, String name) {
+    private static List<CountryDTO> getListOfCountriesNameEquals(List<CountryDTO> allCountries, String name) {
         List<CountryDTO> filteredCountries = new ArrayList<>();
-//    Замість мого коментаря повинна бути реалізація
+
+        for (CountryDTO country : allCountries) {
+            if (country.getName().equalsIgnoreCase(name)) {
+                filteredCountries.add(country);
+            }
+        }
+        //    Замість мого коментаря повинна бути реалізація
 //    Задача 4: знайди всі країни в яких назва(name) дорівнює введеному значенню, наприклад italy.
 //    Пошук повинен бути case insensitive - тобто буде працювати однаково для italy, ITALY і Italy
         return filteredCountries;
     }
 
-    private List<CountryDTO> getListOfCountriesNameContains(List<CountryDTO> allCountries, String name) {
+    private static List<CountryDTO> getListOfCountriesNameContains(List<CountryDTO> allCountries, String name) {
         List<CountryDTO> filteredCountries = new ArrayList<>();
-//    Замість мого коментаря повинна бути реалізація
+
+        for (CountryDTO country : allCountries) {
+
+            if (country.getName().toUpperCase().contains(name.toUpperCase())) {
+                filteredCountries.add(country);
+            }
+        }
+
+        //    Замість мого коментаря повинна бути реалізація
 //    Задача 5: знайди всі країни в яких назва(name) містить введене значення, наприклад united.
 //    Пошук повинен бути case insensitive - тобто буде працювати однаково для united, UNITED і United
         return filteredCountries;
     }
 
-    private List<CountryDTO> getListOfCountriesByName(List<CountryDTO> allCountries, String name, boolean isFullMatch) {
+
+
+    private static List<CountryDTO> getListOfCountriesByName(List<CountryDTO> allCountries, String name, boolean isFullMatch) {
         List<CountryDTO> filteredCountries = new ArrayList<>();
-//    Замість мого коментаря повинна бути реалізація
+
+        return (isFullMatch) ? getListOfCountriesNameEquals(allCountries,name) : getListOfCountriesNameContains(allCountries,name);
```

Comment:
you don't need use round temples for isFullMatch, it should look just like `return isFullMatch ? getListOfCountriesNameEquals(allCountries,name) : getListOfCountriesNameContains(allCountries,name);`